### PR TITLE
[Discovery Server] Remove change in History comparing instance handle

### PR DESCRIPTION
**Pull request by jparisu**
The removing of Data in a History was compared by the guid, but not by the instance Handle.
Different minor fixes.

### CODE REVIEW

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 1]
```diff
@@ -402,9 +411,12 @@ void DiscoveryDataBase::create_participant_from_change_(
         const DiscoveryParticipantChangeData& change_data)
 {
     DiscoveryParticipantInfo part(ch, server_guid_prefix_, change_data);
+    // must be acked also by the entity that has sent the data (in local entities, itself)
+    part.add_or_update_ack_participant(ch->writerGUID.guidPrefix, true);
     std::pair<std::map<eprosima::fastrtps::rtps::GuidPrefix_t, DiscoveryParticipantInfo>::iterator, bool> ret =
             participants_.insert(std::make_pair(guid_from_change(ch).guidPrefix, part));
 
+
     // If insert resturn false, means that the participant already existed (DATA(p) is an update). In that case
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:420]:
```suggestion
    // If insert returns false, means that the participant already existed (DATA(p) is an update). In that case
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 2]
```diff
@@ -402,9 +411,12 @@ void DiscoveryDataBase::create_participant_from_change_(
         const DiscoveryParticipantChangeData& change_data)
 {
     DiscoveryParticipantInfo part(ch, server_guid_prefix_, change_data);
+    // must be acked also by the entity that has sent the data (in local entities, itself)
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:414]:
```suggestion
    // The change must be acked also by the entity that has sent the data (in local entities, itself)
```

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp [hunk 3]
```diff
@@ -64,7 +64,7 @@ class DiscoveryDataBase
 
     class AckedFunctor
     {
-        using argument_type = eprosima::fastrtps::rtps::ReaderProxy*;
+        using argument_type = eprosima::fastrtps::rtps::ReaderProxy *;
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp:67]: Uncrustify manually
```suggestion
        using argument_type = eprosima::fastrtps::rtps::ReaderProxy*;
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 4]
```diff
@@ -32,7 +32,7 @@ namespace ddb {
 DiscoveryDataBase::DiscoveryDataBase(
         fastrtps::rtps::GuidPrefix_t server_guid_prefix)
     : server_guid_prefix_(server_guid_prefix)
-    , server_acked_by_all_(false)
+    , server_acked_by_all_(true)
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:35]: :+1:

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp (hunk 1 above)
Comment (jparisu) [reply src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:420]: :+1:

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp (hunk 2 above)
Comment (jparisu) [reply src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:414]: :+1:

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp (hunk 3 above)
Comment (jparisu) [reply src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.hpp:67]: :cry:

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2876/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/179/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2631/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2887/)

File: src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp [hunk 5]
```diff
@@ -402,10 +411,13 @@ void DiscoveryDataBase::create_participant_from_change_(
         const DiscoveryParticipantChangeData& change_data)
 {
     DiscoveryParticipantInfo part(ch, server_guid_prefix_, change_data);
+    // The change must be acked also by the entity that has sent the data (in local entities, itself)
+    part.add_or_update_ack_participant(ch->writerGUID.guidPrefix, true);
     std::pair<std::map<eprosima::fastrtps::rtps::GuidPrefix_t, DiscoveryParticipantInfo>::iterator, bool> ret =
             participants_.insert(std::make_pair(guid_from_change(ch).guidPrefix, part));
 
-    // If insert resturn false, means that the participant already existed (DATA(p) is an update). In that case
+
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/database/DiscoveryDataBase.cpp:419]:
```suggestion
```

File: src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp [hunk 6]
```diff
@@ -667,13 +667,18 @@ bool PDPServer2::server_update_routine()
     {
         logInfo(RTPS_PDP_SERVER, "");
         logInfo(RTPS_PDP_SERVER, "-------------------- Server routine start --------------------");
+
         process_writers_acknowledgements();     // server + ddb(functor_with_ddb)
         process_data_queues();                  // all ddb
-        process_disposals();                    // server + ddb(changes_to_dispose(), clear_changes_to_disposes())
         process_dirty_topics();                 // all ddb
-        process_to_send_lists();                // server + ddb(get_to_send, remove_to_send_this)
+
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp:674]:
```suggestion
```

File: src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp [hunk 7]
```diff
@@ -667,13 +667,18 @@ bool PDPServer2::server_update_routine()
...
         process_changes_release();              // server + ddb(changes_to_release(), clear_changes_to_release())
+
+        process_disposals();                    // server + ddb(changes_to_dispose(), clear_changes_to_disposes())
+        process_to_send_lists();                // server + ddb(get_to_send, remove_to_send_this)
+
```
Comment (rsanchez15) [inline src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp:679]:
```suggestion
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**